ignite/handlers/checkpoint.py

```diff
@@ -347,7 +355,7 @@ def reset(self) -> None:
         self._saved = []
 
     @property
-    def last_checkpoint(self) -> Optional[str]:
+    def last_checkpoint(self) -> Optional[Union[str, Path]]:
```

Comment:
```suggestion
    def last_checkpoint(self) -> Optional[str]:
```
Actually, filename is a str and not Path